aspnet/migration/http-modules.rst

```diff
@@ -1,8 +1,576 @@
+has changed greatly in ASP.NET 5. `Later on <#migrating-to-the-new-httpcontext>`_, you'll see how to migrate to the new ASP.NET 5 HttpContext.
+
+Migrating module insertion into the request pipeline
+-----------------------------------------------------
+HTTP modules are typically added to the request pipeline using web.config:
+
+.. literalinclude:: http-modules/sample/Asp.Net4/Asp.Net4/Web.config
+	:language: xml
+	:linenos:
+	:emphasize-lines: 6
+	:lines: 1-3, 29-30, 33, 40, 48, 105
+
+Convert this by `adding your new middleware <../fundamentals/middleware.html#creating-a-middleware-pipeline-with-iapplicationbuilder>`_
+to the request pipeline in your ``Startup`` class:
+
+.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 1-2, 12-15, 46-48, 54-58, 105, 109, 110
+	:emphasize-lines: 12
+
+The exact spot in the pipeline where you insert your new middleware depends on the event 
+that it handled as a module (``BeginRequest``, ``EndRequest``, etc.) and its order in your list of modules in your web.config.
+
+However, as you saw there is no more application life cycle in ASP.NET 5
+and the order in which responses are processed by middleware differs from the order used by modules.
+This could make your ordering decision more challenging.
+
+If ordering becomes a problem, you could split your module into multiple middleware that can then be ordered separately.
+
+Migrating handler code to middleware
+-------------------------------------
+An HTTP handler looks something like this:
+
+.. literalinclude:: http-modules/sample/Asp.Net4/Asp.Net4/HttpHandlers/ReportHandler.cs
+	:language: c#
+	:linenos:
+	:emphasize-lines: 5, 7, 13-16
+	:lines: 1-19, 31-32
+
+In your ASP.NET 5 project, you would translate this to a middleware similar to this:
+
+.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/ReportHandlerMiddleware.cs
+	:language: c#
+	:linenos:
+	:emphasize-lines: 7,9,13, 20-23, 29, 31, 33
+	:lines: 1-26, 38-47
+
+This middleware is very similar to the middleware corresponding to modules. The only real difference is that here there is no
+call to ``_next.Invoke(context)``. That makes sense, because the handler is at the end of the request pipeline, so there will be no next middleware to invoke.
+
+Migrating handler insertion into the request pipeline
+------------------------------------------------------
+Configuring an HTTP handler is done in web.config and looks something like this:
+
+.. literalinclude:: http-modules/sample/Asp.Net4/Asp.Net4/Web.config
+	:language: xml
+	:linenos:
+	:emphasize-lines: 7,8
+	:lines: 1-3, 29, 42-46, 48, 105
+
+You could convert this by adding your new handler middleware to the request pipeline in your ``Startup`` class, similar to 
+middleware converted from modules.
+
+Problem is that this would send all requests to your new handler middleware. However, you only want requests with a given
```

Comment:
You could convert this by adding your new handler middleware to the request pipeline in your Startup class, similar to middleware converted from modules. The problem with that approach;  it would send all requests to your new handler middleware. However, you only want requests with a given extension to reach your middleware. That would give you the same functionality you had with your HTTP handler.